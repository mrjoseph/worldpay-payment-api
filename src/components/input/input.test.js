import React from 'react';
import { shallow, configure } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import Input from './input';

configure({ adapter: new Adapter() });

describe('Input', () => {
  const props = {
    type: '',
    className: '',
    name: '',
    placeholder: '',
    autoComplete: '',
    required: '',
    label: '',
    onChange: () => {}
  };

  let component;
  let onChange;
  beforeEach(() => {
    onChange = jest.fn(props.onChange);
    const newProps = {...props, onChange };
    component = shallow (<Input {...newProps} />)

  });

  it('Should call onChange function passed as a prop', () => {
    const input = component.find('input');
    const expected = {
      target: 'hello world'
    };
    expect(onChange.mock.calls.length).toBe(0);
    input.simulate('change', expected);
    expect(onChange).toHaveBeenCalledWith(expected);
    expect(onChange.mock.calls.length).toBe(1);
  });

  it('Should render the input with props', () => {
    const rendered = renderer.create(
        <Input {...props} />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
