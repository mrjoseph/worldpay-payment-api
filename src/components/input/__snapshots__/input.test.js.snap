// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Should render the input with props 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor=""
  >
    
  </label>
  <input
    autoComplete=""
    className=""
    name=""
    onChange={[Function]}
    placeholder=""
    required=""
    type=""
  />
</div>
`;
